frontends/web/src/routes/device/bitbox02/sdcardcheck.tsx

```diff
@@ -14,80 +14,68 @@
  * limitations under the License.
  */
 
-import { Component } from 'react';
+import { FunctionComponent, useEffect, useState } from 'react';
 import { translate, TranslateProps } from '../../../decorators/translate';
 import { Dialog, DialogButtons } from '../../../components/dialog/dialog';
 import { apiGet } from '../../../utils/request';
 import { Button, ButtonLink } from '../../../components/forms';
 
-interface SDCardCheckProps {
+type SDCardCheckProps = {
     deviceID: string;
 }
 
-interface State {
-    sdCardInserted?: boolean;
-}
-
-
 type Props = SDCardCheckProps & TranslateProps;
 
-class SDCardCheck extends Component<Props, State> {
-  state = {} as State;
+const SDCardCheck: FunctionComponent<Props> = ({ t, deviceID, children }) => {
 
-  public componentDidMount() {
-    this.check();
-  }
+  const [sdCardInserted, setSdCardInserted] = useState<boolean | undefined>(undefined);
+
+  useEffect(() => {
+    check();
+    // no dependencies passed - only gets executed once on mount
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Shouldn't the effect re-run if `deviceID` changes, I guess useCallback, but see my later comment about `useLoad`.

Reply:
Didn't get the impression that it should re-run when `deviceID` changes, but anyhow I agree that useLoad is cleaner!

However, if we were to use `useLoad` I wonder what should the `onClick` be for the "ok button"? Because atm, it's firing `check` which affects the local state, which affects the conditional rendering. 

Using `useLoad`, we won't need to use the local state (`inserted`) anymore as the hook returns the value already, but this means we can't control the state on a button click, like we have right now.

Ofc the `onClick` could be just checkSDCard() (API call), but that doesn't really do anything in regards to the UI rendering - so it's pointless at that point, or we could use useEffect+useCallback but this seems like it just overcomplicates things without adding real value. Example:

```js
  const [inserted, setInserted] = useState<boolean | undefined>();
  const checkInserted = useLoad(() => checkSDCard(deviceID), [deviceID]);
  const check = useCallback(() => checkSDCard(deviceID).then(inserted => setInserted(inserted)), [deviceID]);

  useEffect(() => {
    setInserted(checkInserted);
  }, [checkInserted]);
```

In this case, `onClick` will be `check` and we use `inserted` for the conditional rendering instead of the returned value from `useLoad` (`checkInserted`) . 😕

Reply:
I changed it to use useCallback + useEffect which still uses the local useState (without useLoad). Please take a look, otherwise we could use useLoad but question above still stands: what should the onClick be for the "ok button" if useLoad is being used (do we need an ok button at all then)?  🤔

Reply:
You are right `useLoad` doesn't work with clicking the button, the way you have it with useCallback + useEffect is good 👍 .

> Didn't get the impression that it should re-run when deviceID changes, but anyhow I agree that useLoad is cleaner!

In practice right now deviceID would not change as we only have support for 1 BB02 at the time, but if we one day allow multiple, it may change. I believe now it's better as it would create a new callback if the id changes. Before the check would only have been executed once. I believe now it is good and the effect runs again if there is a new check function / id changed.